rpkg_src/decrypt_packagedefinition_thumbs.cpp

```diff
@@ -73,7 +79,18 @@ void generic_function::decrypt_packagedefinition_thumbs(std::string& input_path,
         }
     }
 
-    std::string output_file_base_name = file::output_path_append(file::get_base_file_name(input_path), output_path);
+	uint32_t table[256];
+	crc32::generate_table(table);
+	uint32_t crc = crc32::update(table, 0, input_data.data(), last_zero_position);
+
+	uint32_t old_crc = 0;
+	memcpy(&old_crc, &checksum_raw.data()[0], sizeof(old_crc));
+
+	std::string output_file_base_name = file::output_path_append(file::get_base_file_name(input_path), output_path);
+
+	if (crc != old_crc) {
+		LOG_AND_EXIT("Could not decrypt " + output_file_base_name + "!\nReason: Checksum mismatch! Exiting...");
+	}
```

Comment:
```suggestion
    uint32_t table[256];
    crc32::generate_table(table);
    uint32_t crc = crc32::update(table, 0, input_data.data(), last_zero_position);

    uint32_t old_crc = 0;
    memcpy(&old_crc, &checksum_raw.data()[0], sizeof(old_crc));

    std::string output_file_base_name = file::output_path_append(file::get_base_file_name(input_path), output_path);

    if (crc != old_crc) {
        LOG_AND_EXIT("Could not decrypt " + output_file_base_name + "!\nReason: Checksum mismatch! Exiting...");
	}
```

Reply:
Blame Visual Studio. Fixed in another commit.